### PR TITLE
[Merged by Bors] - feat port: Order.Monovary

### PR DESCRIPTION
**Pull request by ChrisHughes24**
ee0c179cd3c8a45aa5bffbf1b41d8dbede452865

### CODE REVIEW

File: Mathlib/Order/Monovary.lean [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+theorem AntivaryOn.empty : AntivaryOn f g ∅ := fun _ => False.elim
+#align antivary_on.empty AntivaryOn.empty
+
+@[simp]
+theorem monovaryOn_univ : MonovaryOn f g univ ↔ Monovary f g :=
+  ⟨fun h _ _ => h trivial trivial, fun h _ _ _ _ hij => h hij⟩
+#align monovary_on_univ monovaryOn_univ
+
+@[simp]
+theorem antivary_on_univ : AntivaryOn f g univ ↔ Antivary f g :=
```
Comment (fpvandoorn) [inline Mathlib/Order/Monovary.lean:78]:
```suggestion
theorem antivaryOn_univ : AntivaryOn f g univ ↔ Antivary f g :=
```
also in align

File: Mathlib/Order/Monovary.lean [hunk 2]
```diff
@@ -45,239 +45,239 @@ def Antivary (f : ι → α) (g : ι → β) : Prop :=
...
-    MonovaryOn (f' ∘ f) g s := fun i hi j hj hij => hf <| h hi hj hij
+    MonovaryOn (f' ∘ f) g s := fun _ hi _ hj hij => hf <| h hi hj hij
 #align antivary_on.comp_antitone_on_left AntivaryOn.comp_antitone_on_left
 
 section OrderDual
 
 open OrderDual
 
-theorem Monovary.dual : Monovary f g → Monovary (to_dual ∘ f) (to_dual ∘ g) :=
+theorem Monovary.dual : Monovary f g → Monovary (toDual ∘ f) (toDual ∘ g) :=
```
Comment (fpvandoorn) [inline Mathlib/Order/Monovary.lean:194]: is `to_dual` missing an align?

File: Mathlib/Order/Monovary.lean [hunk 3]
```diff
@@ -310,39 +310,39 @@ end PartialOrder
 variable [LinearOrder ι]
 
 protected theorem Monotone.monovary (hf : Monotone f) (hg : Monotone g) : Monovary f g :=
-  fun i j hij => hf (hg.reflect_lt hij).le
+  fun _ _ hij => hf (hg.reflect_lt hij).le
 #align monotone.monovary Monotone.monovary
 
 protected theorem Monotone.antivary (hf : Monotone f) (hg : Antitone g) : Antivary f g :=
-  (hf.Monovary hg.dual_right).dual_right
+  (hf.monovary (Antitone.dual_right hg)).dual_right
```
Comment (fpvandoorn) [inline Mathlib/Order/Monovary.lean:317]: `hg.dual_right` doesn't work?

Comment (fpvandoorn) [reply Mathlib/Order/Monovary.lean:317]: can you fix
`alias monotone_toDual_comp_iff ↔ _ Antitone.dual_right`
at
https://github.com/leanprover-community/mathlib4/blob/be3b492c47a934ea2fda7307316b6869b91944d8/Mathlib/Order/Monotone/Basic.lean#L244-L244

Comment (fpvandoorn) [reply Mathlib/Order/Monovary.lean:317]: and similar for all other projection notations that are not working?

Comment (ChrisHughes24) [reply Mathlib/Order/Monovary.lean:317]: I made a [Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60alias.60.20unfolds.20definitions). This file is low priority. Maybe `alias` is easy to fix.

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**